wallet/src/Pos/Wallet/Web/Pending/Worker.hs

```diff
@@ -0,0 +1,174 @@
+    ( startPendingTxsResubmitter
+    ) where
+
+import           Universum
+
+import           Control.Monad.Catch          (handleAll)
+import           Data.Time.Units              (Second, convertUnit)
+import           Formatting                   (build, sformat, shown, (%))
+import           Mockable                     (delay, fork)
+import           Serokell.Util.Text           (listJson)
+import           System.Wlog                  (logError, logInfo, modifyLoggerName)
+
+import           Pos.Client.Txp.Addresses     (MonadAddresses)
+import           Pos.Communication            (submitAndSave)
+import           Pos.Communication.Protocol   (SendActions (..))
+import           Pos.Core                     (FlatSlotId, SlotId (..), getSlotCount)
+import           Pos.Core.Slotting            (flattenSlotId)
+import           Pos.Crypto                   (WithHash (..))
+import           Pos.Slotting                 (getLastKnownSlotDuration, onNewSlot)
+import           Pos.Txp                      (ToilVerFailure (..), TxAux (..),
+                                               getMemPool, processTx, runDBToil,
+                                               runToilTLocal, topsortTxs)
+import qualified Pos.Wallet.Web.Mode
+import           Pos.Wallet.Web.Pending.Types (PendingTx (..), PtxCondition (..))
+import           Pos.Wallet.Web.State         (casPtxCondition, getPendingTxs)
+
+type MonadPendings m =
+    ( m ~ Pos.Wallet.Web.Mode.WalletWebMode
+    , MonadAddresses m  -- TODO [CSM-407]: for now there is no way to know
+                        -- about this instance here
+    )
+
+-- | What should happen with pending transaction if attempt
+-- to resubmit it failed.
+processPtxFailure :: MonadPendings m => PendingTx -> ToilVerFailure -> m ()
+processPtxFailure PendingTx{..} e =
+    -- If number of 'ToilVerFailure' constructors will ever change, compiler
+    -- will complain - for this purpose we consider all cases explicitly here.
+    case e of
+        ToilKnown                -> tryLater
+        ToilTipsMismatch{}       -> tryLater
+        ToilSlotUnknown          -> tryLater
+        ToilOverwhelmed{}        -> tryLater
+        ToilNotUnspent{}         -> discard
+        ToilOutGTIn{}            -> discard
+        ToilInconsistentTxAux{}  -> discard
+        ToilInvalidOutputs{}     -> discard
+        ToilInvalidInputs{}      -> discard
+        ToilTooLargeTx{}         -> discard
+        ToilInvalidMinFee{}      -> discard
+        ToilInsufficientFee{}    -> discard
+        ToilUnknownAttributes{}  -> discard
+        ToilBootDifferentStake{} -> discard
+  where
+    tryLater = pass
+    discard  = do
+        casPtxCondition ptxTxId PtxApplying (PtxWon'tApply $ sformat build e)
+        logInfo $ sformat ("Transaction "%build%" was canceled") ptxTxId
+
+filterApplicablePtxs
+    :: MonadPendings m
+    => SlotId -> [PendingTx] -> m [PendingTx]
+filterApplicablePtxs curSlot ptxs = do
```

Comment:
Am I right that I have to take block semaphore here? Otherwise I can catch a moment when some conflicting transaction is removed from mempool but not added to blocks yet.
If I have to, won't it be too expensive?